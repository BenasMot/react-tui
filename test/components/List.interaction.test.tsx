import React from "react";
import { createSceenContext, TestSceenContext } from "../fixtures/context";
import { Box, List } from "../../src/components/primitives";

describe("aaa", () => {
  let context: TestSceenContext;
  beforeEach(() => {
    context = createSceenContext();
  });

  test("<List />", async (t) => {
    const {
      render,
      RootBox,
      screen,
      screenToString,
      settle,
    } = context;
    let swapChild: (v: boolean) => void = (v) => {};
    function SwappingRootChild() {
      const [isChildSwapped, _swapChild] = React.useState(false);
      swapChild = _swapChild;
      return (
        <RootBox>
          {isChildSwapped ? <Box>swapped</Box> : (
            <List
              label="some-list"
              border={{ type: "line" }}
              style={{ border: { fg: "blue" } }}
              items={["a", "b", "c"]}
            />
          )}
        </RootBox>
      );
    }
    /**
     * observe us set items in a list, swap them out, and swap the list back again
     */
    await render(<SwappingRootChild />);
    await settle();
    expect(screenToString()).toMatchSnapshot();
    swapChild(true);
    await settle();
    screen.options?.input?.write("\x1b"); // send an ESCape key! mother trucker!!
    expect(screenToString()).toMatchSnapshot();
    swapChild(false);
    await settle();
    expect(screenToString()).toMatchSnapshot();
  });
});
