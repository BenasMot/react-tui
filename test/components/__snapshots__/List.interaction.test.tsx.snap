// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aaa <List /> 1`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─some-list──────────────────────────────────────────────┐│
││a                                                       ││
││b                                                       ││
││c                                                       ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;

exports[`aaa <List /> 2`] = `
"┌─root-box─────────────────────────────────────────────────┐
│swapped                                                   │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
│                                                          │
└──────────────────────────────────────────────────────────┘"
`;

exports[`aaa <List /> 3`] = `
"┌─root-box─────────────────────────────────────────────────┐
│┌─some-list──────────────────────────────────────────────┐│
││a                                                       ││
││b                                                       ││
││c                                                       ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘"
`;
